src/Microsoft.Diagnostics.EventFlow.Inputs.EventSource/EventSourceInput.cs

```diff
@@ -155,46 +158,95 @@ private void EnableAsNecessary(EventSource eventSource)
                 }
             }
         }
 
-        private void Initialize(IReadOnlyCollection<EventSourceConfiguration> eventSources, IHealthReporter healthReporter)
+        private void Initialize(List<EventSourceConfiguration> eventSources, IHealthReporter healthReporter)
         {
             this.healthReporter = healthReporter;
             this.subject = new EventFlowSubject<EventData>();
 
+            if (eventSources.Count() == 0)
+            {
+                healthReporter.ReportWarning($"{nameof(EventSourceInput)}: no event sources configured", EventFlowContextIdentifiers.Configuration);                
+            }
+
+            var invalidConfigurationItems = new List<EventSourceConfiguration>();
+            foreach(var eventSourceConfiguration in eventSources)
+            {
+                if (!eventSourceConfiguration.Validate())
+                {
+                    healthReporter.ReportProblem($"{nameof(EventSourceInput)}: configuration for one of the sources is invalid", EventFlowContextIdentifiers.Configuration);
+                    invalidConfigurationItems.Add(eventSourceConfiguration);
+                }
+            }
+            // eventSources is a collection created by us, so we can modify it as necessary
+            eventSources.RemoveAll(config => invalidConfigurationItems.Contains(config));
             this.EventSources = eventSources;
-            if (this.EventSources.Count() == 0)
+
+            bool haveDisabledSources = this.EventSources.Any(config => !string.IsNullOrWhiteSpace(config.DisabledProviderNamePrefix));
+            if (haveDisabledSources)
             {
-                healthReporter.ReportWarning($"{nameof(EventSourceInput)}: no event sources configured", nameof(EventSourceInput));                
+                this.disabledSources = new ConcurrentDictionary<string, bool>();
+                this.OnEventWrittenImpl = BroadcastEventIfSourceNotDisabled;
             }
-            
+            else
+            {
+                this.OnEventWrittenImpl = BroadcastEvent;
+            }
+
             // Make sure the constructor has run to completion before enabling any sources.
             this.initialization = Task.Run(() =>
             {
                 this.constructed = true;
                 EnableInitialSources();
             });
         }
+
+        private void BroadcastEventIfSourceNotDisabled(EventWrittenEventArgs eventArgs)
+        {
+            string eventSourceName = eventArgs.EventSource.Name;
+            if (eventSourceName != null)
+            {
+                bool isDisabled;
+                if (!this.disabledSources.TryGetValue(eventSourceName, out isDisabled))
+                {
+                    isDisabled = this.EventSources.Any(eventSourceConfiguration => eventSourceConfiguration.Disables(eventArgs.EventSource));
+                    this.disabledSources[eventSourceName] = isDisabled;
```

Comment:
Maybe I don't remember the architecture of eventflow. Instead of checking this on every event, can't it work like, the provider gets disabled up front, and you won't see its events at all?

Reply:
That is what the issue in the documentation is about--with this bug we are getting events even if we do not enable the provider (EventSource) at all. So we need to check where every events is coming from

Reply:
Ah ok